.github/workflows/haskell-ci.yml

```diff
@@ -1,82 +1,116 @@
 # This GitHub workflow config has been generated by a script via
 #
-#   haskell-ci 'github' '--config=cabal.haskell-ci' 'cabal.project'
+#   haskell-ci '--distribution' 'jammy' 'github' 'cabal.project'
 #
 # To regenerate the script (for example after adjusting tested-with) run
 #
 #   haskell-ci regenerate
 #
 # For more information, see https://github.com/haskell-CI/haskell-ci
 #
-# version: 0.12.1
+# version: 0.16.3
 #
-# REGENDATA ("0.12.1",["github","--config=cabal.haskell-ci","cabal.project"])
+# REGENDATA ("0.16.3",["--distribution","jammy","github","cabal.project"])
 #
 name: Haskell-CI
 on:
-  push:
-    branches:
-      - master
-  pull_request:
-    types:
-      - opened 
-      - reopened 
-      - synchronize
-      - ready_for_review
+  - push
+  - pull_request
 jobs:
   linux:
-    if: github.event.pull_request.draft == false
     name: Haskell-CI - Linux - ${{ matrix.compiler }}
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-22.04
+    timeout-minutes:
+      60
     container:
-      image: buildpack-deps:bionic
+      image: buildpack-deps:jammy
     continue-on-error: ${{ matrix.allow-failure }}
     strategy:
       matrix:
         include:
           - compiler: ghc-9.0.1
```

Comment:
Is there a reason we are still sticking to an old GHC version? I think 9.6.2 is the latest stable available. We should update the CI to use the latest available version.

Reply:
Well, Gibbon as it is on the master branch cannot be built by newer GHCs because of this upper bound on base: https://github.com/iu-parfunc/gibbon/blob/83d539443d17584f11daec9a4eaffe193e6929e8/gibbon-compiler/gibbon.cabal#L87

I'll update it 9.6 later today.

Reply:
Ah, we need base version 4.18. 
SG.

Reply:
Note that not all HLS versions are supported the same, for example with [tooling](https://haskell-language-server.readthedocs.io/en/latest/support/ghc-version-support.html#current-ghc-version-support-status). 9.0.2, 9.2.7, 9.4.5, 9.6.1 may be more supported in general by the ecosystem, so maybe we should land on one of those.

Reply:
I've updated Gibbon to use GHC v9.6.2. @vidsinghal @jazullo Does it look ok to merge now?